fix(core): wrong tile transformation

gchoqueux opened the pull request:
stop use root layer transformation to compute tile transformation
especially for the animation of the root

fix #668 and #642

peppsac commented on src/Core/Scheduler/Providers/TileProvider.js:
```diff
@@ -95,11 +94,11 @@ TileProvider.prototype.executeCommand = function executeCommand(command) {
-        parent.worldToLocal(position);
-        worldQuaternion.setFromRotationMatrix(parent.matrixWorld).inverse().multiply(layer.object3d.quaternion);
-        quaternion.premultiply(worldQuaternion);
+    tile.extentTransform = { quaternionInv: quaternion.clone().inverse(), position: position.clone() };
```
At the very least add a `_` prefix.
But storing 2 extra properties, only to be able to compute the transform of children is not necessary.

Here's a version that also fixes the issue:
```js
    if (parent) {
        // position / quaternion are initially in GIS coordinates
        const p1 = new THREE.Vector3();
        const p2 = p1.clone();

        position.applyMatrix4(layer.object3d.matrixWorld);

        layer.object3d.matrixWorld.decompose(p1, worldQuaternion, p2);
        quaternion.premultiply(worldQuaternion);

        // Now position / quaternion are in World coordinates
        // But we want them in parent coordinates
        const m = new THREE.Matrix4();
        m.getInverse(parent.matrixWorld);
        parent.matrixWorld.decompose(p1, worldQuaternion, p2);

        // First, transform position
        position.applyMatrix4(m);
        // Second, transform quaternion
        worldQuaternion.inverse();
        quaternion.premultiply(worldQuaternion);
    }
```

(we have to use `decompose` and not `setFromRotationMatrix` because the latter is broken when scale is not 1)

gchoqueux replied:
I also wanted to stop the use of `layer.object3d` which does not make much sense here

We can do this

```js
if (parent && parent instanceof TileMesh) {
        // get parent extent transformation
        const pTrans = builder.computeSharableExtent(parent.extent);
        // place relative to his parent
        position.sub(pTrans.position).applyQuaternion(pTrans.quaternion.inverse());
        quaternion.premultiply(pTrans.quaternion);
}
```

peppsac replied:
> layer.object3d which does not make much sense here`

Using `layer.object3d` is mandatory if you use `parent.matrixWorld`, because `parent` is a descendant of `layer.object3d`.

But your alternative solution is looking good too.

gchoqueux replied:
later I will calculate the position and quaternion relative to the parent in `computeSharableExtent`

gchoqueux committed:
fix(core): use parent extent transformation to compute tile's transformation

stop use root layer transformation to compute tile transformation
especially for the animation of the root

fix #668 and #642

peppsac commented:
Looks good. Can you squash the commits and change the commit message?

gchoqueux commented:
what commit message is not suitable?

peppsac commented:
` fix(core): wrong tile transformation`: should summarize the content of the commit and must start with a verb.
And you could squash the 2nd commit in the first.

gchoqueux commented:
done

peppsac commented:
> done

You forgot to squash the 2nd commit?

gchoqueux commented:
squashed
